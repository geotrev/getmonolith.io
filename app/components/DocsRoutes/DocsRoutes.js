import React from "react"
import { Switch, Route } from "react-router-dom"
import Loadable from "react-loadable"

import * as routes from "app/routes"
import PageNotFound from "app/components/PageNotFound"
import LoadingSpinner from "app/components/LoadingSpinner"

const Loading = () => <LoadingSpinner />
Loading.displayName = "Loading"

function getRoute(name) {
  return Loadable({
    loader: () => import(`./articles/${name}`),
    loading: Loading,
  })
}

export default function DocsRoutes() {
  return (
    <Switch>
      <Route exact path={routes.introductionPath} component={getRoute("introduction")} />
      <Route exact path={routes.downloadPath} component={getRoute("download")} />
      <Route exact path={routes.brandingPath} component={getRoute("branding")} />
      <Route exact path={routes.javascriptPath} component={getRoute("javascript")} />
      <Route exact path={routes.gridPath} component={getRoute("grid")} />
      <Route exact path={routes.typographyPath} component={getRoute("typography")} />
      <Route exact path={routes.buttonsPath} component={getRoute("buttons")} />
      <Route exact path={routes.formsPath} component={getRoute("forms")} />
      <Route exact path={routes.modalsPath} component={getRoute("modals")} />
      <Route exact path={routes.accordionsPath} component={getRoute("accordions")} />
      <Route exact path={routes.dropdownsPath} component={getRoute("dropdowns")} />
      <Route exact path={routes.tooltipsPath} component={getRoute("tooltips")} />
      <Route exact path={routes.alignmentPath} component={getRoute("alignment")} />
      <Route exact path={routes.offsetOrderPath} component={getRoute("offset-order")} />
      <Route exact path={routes.textPath} component={getRoute("text")} />
      <Route exact path={routes.displayPath} component={getRoute("display")} />
      <Route exact path={routes.colorPath} component={getRoute("color")} />
      <Route exact path={routes.spacingPath} component={getRoute("spacing")} />
      <Route component={PageNotFound} />
    </Switch>
  )
}
