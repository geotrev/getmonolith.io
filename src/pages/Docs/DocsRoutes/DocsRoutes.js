import React, { Component } from "react"
import { Route } from "react-router-dom"

import { Article, HeaderText } from "components"
import { Link } from "react-router-dom"
import ChevronRight from "react-feather/dist/icons/chevron-right"
import { ScrollUpOnMount } from "helpers"

// import OverviewMd from "articles/Home.md"
// import GettingStartedMd from "articles/Getting-Started.md"
// import ConfigMd from "articles/Configuration.md"
// import GridMd from "articles/Grid.md"
// import TypeMd from "articles/Typography.md"
// import ButtonsMd from "articles/Buttons.md"
// import FormsMd from "articles/Forms.md"
// import StyleUtilitiesMd from "articles/Style-Utilities.md"
// import JSUtilitiesMd from "articles/JS-Utilities.md"
// import ModalsMd from "articles/Modals.md"
// import AccordionsMd from "articles/Accordions.md"

const Template = props => {
  return <div className="articles-wrapper row">{props.children}</div>
}

export default class DocsRoutes extends Component {
  render() {
    return (
      <div className="docs-routes-wrapper small-section grid">
        <div className="row">
          {/* <Route
            exact
            path="/docs/overview"
            render={() => this.renderArticle(OverviewMd, "Overview")}
          />
          <Route
            exact
            path="/docs/getting-started"
            render={() => this.renderArticle(GettingStartedMd, "Getting Started")}
          />
          <Route
            exact
            path="/docs/configuration"
            render={() => this.renderArticle(ConfigMd, "Configuration")}
          />
          <Route exact path="/docs/grid" render={() => this.renderArticle(GridMd, "Grid", true)} />
          <Route
            exact
            path="/docs/typography"
            render={() => this.renderArticle(TypeMd, "Typography", true)}
          />
          <Route
            exact
            path="/docs/buttons"
            render={() => this.renderArticle(ButtonsMd, "Buttons", true)}
          />
          <Route
            exact
            path="/docs/forms"
            render={() => this.renderArticle(FormsMd, "Forms", true)}
          />
          <Route
            exact
            path="/docs/modals"
            render={() => this.renderArticle(ModalsMd, "Modals", true)}
          />
          <Route
            exact
            path="/docs/accordions"
            render={() => this.renderArticle(AccordionsMd, "Accordions", true)}
          />
          <Route
            exact
            path="/docs/style-utilities"
            render={() => this.renderArticle(StyleUtilitiesMd, "Style Utilities")}
          />
          <Route
            exact
            path="/docs/javascript-utilities"
            render={() => this.renderArticle(JSUtilitiesMd, "JavaScript Utilities")}
          /> */}
        </div>
      </div>
    )
  }

  renderArticle(article, pageName = "", hasExample = false) {
    return (
      <div className="column">
        <ScrollUpOnMount />
        <HeaderText>{pageName}</HeaderText>
        {hasExample && this.renderExampleLink(pageName.toLowerCase())}
        <Template>
          <Article>{article}</Article>
        </Template>
      </div>
    )
  }

  renderExampleLink(page) {
    return (
      <Link className="small secondary button has-feather" to={`/examples/${page}`}>
        See Examples <ChevronRight size={16} />
      </Link>
    )
  }
}
