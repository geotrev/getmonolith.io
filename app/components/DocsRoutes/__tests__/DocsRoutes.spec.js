import React from "react"
import { BrowserRouter as Router } from "react-router-dom"
import DocsRoutes from "../DocsRoutes"

function mountRoutes() {
  return mount(
    <Router>
      <DocsRoutes />
    </Router>
  )
}

describe("<DocsRoutes />", () => {
  it("matches snapshot", () => {
    const wrapper = mountRoutes()
    expect(wrapper).toMatchSnapshot()
  })
})
