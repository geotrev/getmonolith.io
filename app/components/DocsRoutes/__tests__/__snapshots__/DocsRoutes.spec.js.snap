// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocsRoutes /> renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DocsRoutes>
      <Switch>
        <Route
          component={[Function]}
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
        >
          <PageNotFound
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="has-padding-4 has-center-text"
            >
              <h1>
                Sorry, that page doesn't exist. :(
              </h1>
              <p
                className="has-no-margin-bottom"
              >
                Did you enter the URL correctly?
              </p>
            </div>
          </PageNotFound>
        </Route>
      </Switch>
    </DocsRoutes>
  </Router>
</BrowserRouter>
`;
